test/types/index.d.ts

```diff
@@ -1 +1 @@
-// Minimum TypeScript Version: 3.1
+// Minimum TypeScript Version: 4.0
```

Comment:
Normally I think there are some consequences with this, but given that we apparently need to release a major version bump anyway, this is probably fine. Though @chriskrycho should probably confirm just in case. (I am not sure how much of the `@glimmer/*` types ended up being consumed directly, as opposed to being replaced/superseded by the Glint wrappers anyway.)

Reply:
This is indeed breaking! And at this point the majority of the ecosystem is not using Glint, in past because addons cannot do so. (That’s one of the big motivations for moving the Signature RFC forward.) If this is intending to target a 2.0 release, the breaking change is fine. Otherwise we should leave it at its previous value.

Reply:
We already have to do a breaking change anyway without doing _an enormous_ (imo) amount of work to support from ember 3.12 to current (macros, etc)

Reply:
That seems fine, then, with two caveats:

1. I believe there are a number of *other* changes already queued up for a v2, so we need to evaluate those as part of this work.
2. We need to make sure that end users can actually successfully have *both* v1 and v2 active in an app's codebase, since it will take some time to migrate the ecosystem.

Two additional considerations:

- We *can* support the Glint Signature changes without a breaking change, but it would be much easier to *not*. We need to decide on what to do with that.
- There are other changes we *want* to make to align the Glimmer.js API with the version of the Glimmer component API available in Ember.js, specifically around lifecycle hooks. I am not sure if those have yet landed, but we need to follow up and evaluate those before we actually publish the v2.

Reply:
what specific things do we need to do to move forward?
(I'm not clear on what next steps are)

Reply:
For the most part, I don't think should affect this PR, considering we already released 17 (!) betas for 2.0.0. But those are probably checklist items for the 2.0.0 final.